cli/src/pcluster/models/cluster.py

```diff
@@ -469,11 +487,11 @@ def compute_instances(self) -> List[ClusterInstance]:
     @property
     def head_node_instance(self) -> ClusterInstance:
         """Get head node instance."""
-        try:
-            instance_data = self._describe_instances(node_type=NodeType.HEAD_NODE)[0]
-            return ClusterInstance(instance_data)
-        except Exception as e:
```

Comment:
would it make sense to continue wrapping this in a try-except so that the message contained in the ClusterActionError always mentions the head node when raised as a result of calling this method?

Reply:
The problem is that ClusterActionError is too generic while in the caller I need to produce different errors/responses if the failure is due to specific AWS errors (e.g. lack of policy) or is due to head node not running. 
Once I migrate other functionalities to the new model I should review this exception handling mechanism from the cluster class.